pkg/operator/ceph/pool/controller.go

```diff
@@ -341,11 +347,31 @@ func (r *ReconcileCephBlockPool) reconcileCreatePool(clusterInfo *cephclient.Clu
 	return reconcile.Result{}, nil
 }
 
+func getCephName(pool cephv1.CephBlockPool) (string, error) {
+	if pool.Name == "" {
+		return "", errors.Errorf("metadata name must be set")
+	}
+	// If the name is not overridden in the pool spec.name, set it to the name of the pool CR
+	if pool.Spec.Name == "" {
+		return pool.Name, nil
+	}
+	if pool.Spec.Name == pool.Name {
+		// The same pool name is set, nothing is being overridden
+		return pool.Name, nil
+	}
+	// Only allow the name to be overridden for certain built-in pool names
+	if pool.Spec.Name == "device_health_metrics" || pool.Spec.Name == ".nfs" {
+		return pool.Spec.Name, nil
+	}
+
+	return "", errors.Errorf("not allowed to override the pool name except for the device_health_metrics and .nfs pools")
+}
+
 // Create the pool
-func createPool(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, clusterSpec *cephv1.ClusterSpec, p *cephv1.CephBlockPool) error {
+func createPool(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, clusterSpec *cephv1.ClusterSpec, p *cephv1.NamedPoolSpec) error {
 	// create the pool
-	logger.Infof("creating pool %q in namespace %q", p.Name, p.Namespace)
-	if err := cephclient.CreatePoolWithProfile(context, clusterInfo, clusterSpec, p.Name, p.Spec, poolApplicationNameRBD); err != nil {
+	logger.Infof("creating pool %q in namespace %q", p.Name, clusterInfo.Namespace)
+	if err := cephclient.CreatePoolWithProfile(context, clusterInfo, clusterSpec, *p, poolApplicationNameRBD); err != nil {
 		return errors.Wrapf(err, "failed to create pool %q", p.Name)
 	}
 
```

Comment:
Should we validate the pool name a second time here to ensure that any internal usage doesn't try to modify the wrong pools via the `Name` field?

Reply:
Not sure what validation would be needed. The `NamedPoolSpec` is intentionally passed here so that the `Name` field will be explicitly used for pool creation. The `CephBlockPool` object was not passed here, so the original k8s name isn't even known at this point.

Reply:
I see. Okay. 👍